### PR TITLE
Regenerate permalink if its a duplicate

### PR DESCRIPTION
**Pull request by exsuzme**
ping @richardnuno

### CODE REVIEW

File: core/lib/spree/core/permalinks.rb [hunk 1]
```diff
@@ -47,25 +47,20 @@ def generate_permalink
         "#{self.class.permalink_prefix}#{Array.new(self.class.permalink_length){rand(9)}.join}"
       end
 
-      def save_permalink(permalink_value=self.to_param)
+      def save_unique_permalink(permalink_value=self.to_param)
         self.with_lock do
+          binding.pry
```
Comment (richardnuno) [inline core/lib/spree/core/permalinks.rb]: This `binding.pry` snuck in

File: core/spec/lib/spree/core/permalinks_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        expect(new_shipment).to receive(:generate_permalink).and_return(new_permalink)
+
+        subject
+        expect(new_shipment.number).to eq(new_permalink)
+      end
+    end
+
+    context "generate_permalink returns a non-existing permalink" do
+      it "permalink is not nil" do
+        expect(subject).to_not be_nil
```
Comment (richardnuno) [inline core/spec/lib/spree/core/permalinks_spec.rb]: I think this should be `expect(subject.number).to_not be_nil`

File: core/lib/spree/core/permalinks.rb (hunk 1 above)
Comment (exsuzme) [inline core/lib/spree/core/permalinks.rb]: oops, fixed :blush:

File: core/spec/lib/spree/core/permalinks_spec.rb (hunk 2 above)
Comment (exsuzme) [inline core/spec/lib/spree/core/permalinks_spec.rb]: subject.number gave me `undefined method "number" for true:TrueClass`,  instead I called subject first and then asserted on the new_shipment object. Let me know if you can think of a better way to handle that!

Comment (gvaughn) [inline core/spec/lib/spree/core/permalinks_spec.rb]: I would consider making the `subject` block return the new_shipment itself.

**Comment by richardnuno** [comment]
:+1:

**Pull request merged**